lib/Provider/MailProvider.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+namespace OCA\Mail\Provider;
+
+use OCA\Mail\Account;
+use OCA\Mail\Service\AccountService;
+use OCP\Mail\Provider\Address as MailAddress;
+use OCP\Mail\Provider\IProvider;
+use OCP\Mail\Provider\IService;
+use Psr\Container\ContainerInterface;
+
+class MailProvider implements IProvider {
+
+	private ?array $ServiceCollection = [];
+
+	public function __construct(
+		protected ContainerInterface $container,
+		protected AccountService $accountService
+	) {
+	}
+
+	/**
+	 * arbitrary unique text string identifying this provider
+	 *
+	 * @since 4.0.0
+	 *
+	 * @return string					id of this provider (e.g. UUID or 'IMAP/SMTP' or anything else)
+	 */
+	public function id(): string {
+		return 'mail-application';
+	}
+
+	/**
+	 * localized human friendly name of this provider
+	 *
+	 * @since 4.0.0
+	 *
+	 * @return string					label/name of this provider (e.g. Plain Old IMAP/SMTP)
+	 */
+	public function label(): string {
+		return 'Mail Application';
+	}
+
+	/**
+	 * determine if any services are configured for a specific user
+	 *
+	 * @since 4.0.0
+	 *
+	 * @param string $userId			system user id
+	 *
+	 * @return bool 					true if any services are configure for the user
+	 */
+	public function hasServices(string $userId): bool {
+		return (count($this->listServices($userId)) > 0);
+	}
+
+	/**
+	 * retrieve collection of services for a specific user
+	 *
+	 * @since 4.0.0
+	 *
+	 * @param string $userId			system user id
```

Comment:
Please merge main and run `composer run cs:fix` to format the code according to our latest coding-standard update.

Reply:
The code style was always like that (c.f. https://docs.nextcloud.com/server/latest/developer_manual/getting_started/codingguidelines.html#comments), new is that code fixer is checking it properly.